tests/unit/recommenders/datasets/test_download_utils.py

```diff
@@ -25,7 +25,7 @@ def test_maybe_download(files_fixtures):
         os.remove(filepath)
```

Comment:
It seems I broke something when I renamed one of the folders
```
"error": ***
        "message": "***\n    \"error_details\": ***\n        \"componentName\": \"project\",\n        \"correlation\": ***\n            \"operation\": \"d7b079d7bbac3dd9c21f12b04[151](https://github.com/recommenders-team/recommenders/actions/runs/9918504884/job/27403242773?pr=2130#step:3:158)b40d\",\n            \"request\": \"cd3d47291294bee4\"\n        ***,\n        \"environment\": \"eastus\",\n        \"error\": ***\n            \"code\": \"ValidationError\",\n            \"details\": [\n                ***\n                    \"code\": \"Invalid\",\n                    \"message\": \"Failed to create snapshot. File node tests/smoke/recommenders/models/__init__.py was specified but does not appear in the list of uploaded files. Please delete the snapshot cache and resubmit. Snapshot cache is located in .projectcontent folder, typically under .azureml folder in your home folder\",\n                    \"target\": \"nodes\"\n                ***\n            ],\n            \"message\": \"Failed to create snapshot. File node tests/smoke/recommenders/models/__init__.py was specified but does not appear in the list of uploaded files. Please delete the snapshot cache and resubmit. Snapshot cache is located in .projectcontent folder, typically under .azureml folder in your home folder\",\n            \"target\": \"nodes\"\n        ***,\n        \"location\": \"eastus\",\n        \"statusCode\": 400,\n        \"time\": \"2024-07-13T08:14:28.3166442+00:00\"\n    ***,\n    \"status_code\": 400,\n    \"url\":
```

Reply:
@SimonYansenZhao do you know what is this snapshot cache?

Reply:
@miguelgfierro After research, I think a snapshot is just a copy of recommenders.  In the testing workflow, the repo is copied onto the GitHub CICD VM by the action checkout:
https://github.com/recommenders-team/recommenders/blob/d333a0d9ee1680aa231905b319edb567839e3d1c/.github/workflows/azureml-unit-tests.yml#L38-L39

then the copy is uploaded onto the AzureML compute by the Azure ML SDK as a snapshot.  I am not sure why the `__init__.py` file wasn't uploaded.  But it is now uploaded into the snapshot cache after I disable snapshot and re-enable.  In other words, as the error suggested, we need to delete the snapshot cache directory.  Disabling snapshot seemed to have the effect of removing the cache directory, and then re-enabling snapshot makes sure recommenders repo is uploaded onto the AzureML compute.  It looks like a bug of AzureML.

Two solutions that may permanently resolve this problem are:
1. Add code to delete the snapshot cache directory every time when uploading the copy of recommenders.
2. Clone recommenders directly in the docker image used for AzureML compute.